packages/connection-router-api/src/primitives.rs

```diff
@@ -162,7 +162,7 @@ impl FromStr for ChainName {
     type Err = Error;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-        if s.contains(ID_SEPARATOR) || s.is_empty() {
+        if s.contains(CHAIN_NAME_DELIMITER) || s.is_empty() {
```

Comment:
just as confirmation of my assumption: it's fine to have "_" in the ID? More specifically, the only thing we restrict for the ID is that it's not empty, any other restriction is chain specific

Reply:
yes that's correct, we just restrict non-empty for the ID and everything else is chain specific